File: src/2025-03-04-03-28-40--deepseek-chat-0304/python/exercises/practice/bottle-song/bottle_song.py
def number_to_words(n):
    """Convert a number to its word representation."""
    num_words = ["Zero", "One", "Two", "Three", "Four", 
                "Five", "Six", "Seven", "Eight", "Nine", "Ten"]
    return num_words[n]

def recite(start, take=1):
    """Generate verses of the 'Ten Green Bottles' song."""
    verses = []
    for i in range(start, start - take, -1):
        current = number_to_words(i)
        next_num = number_to_words(i-1) if i > 1 else "no"
        
        verse = [
            f"{current} green bottle{'s' if i != 1 else ''} hanging on the wall,",
            f"{current} green bottle{'s' if i != 1 else ''} hanging on the wall,",
            "And if one green bottle should accidentally fall,",
            f"There'll be {next_num} green bottle{'s' if i-1 != 1 else ''} hanging on the wall."
        ]
        
        verses.extend(verse)
        if i > start - take + 1:
            verses.append("")
    
    return verses

File: src/2025-03-04-03-28-40--deepseek-chat-0304/java/exercises/practice/bottle-song/src/main/java/BottleSong.java
/**
 * The BottleSong class generates the lyrics for the "Ten Green Bottles" song.
 * It can generate verses starting from any number of bottles and for any number of verses.
 */
public class BottleSong {

    /**
     * Generates a single verse for the given number of bottles.
     * @param bottles The number of bottles in the verse (must be >= 0)
     * @return The verse as a formatted string
     * @throws IllegalArgumentException if bottles is negative
     */
    private String getVerse(int bottles) {
        if (bottles < 0) {
            throw new IllegalArgumentException("Number of bottles cannot be negative");
        }
        
        if (bottles == 0) {
            return "No green bottles hanging on the wall,\n" +
                   "No green bottles hanging on the wall,\n" +
                   "There are no green bottles hanging on the wall.\n";
        }
        
        String bottleWord = bottles == 1 ? "bottle" : "bottles";
        String nextBottleWord = (bottles - 1) == 1 ? "bottle" : "bottles";
        String nextCount = bottles - 1 == 0 ? "no" : String.valueOf(bottles - 1);
        
        return String.format("%d green %s hanging on the wall,\n" +
                           "%d green %s hanging on the wall,\n" +
                           "And if one green bottle should accidentally fall,\n" +
                           "There'll be %s green %s hanging on the wall.\n",
                           bottles, bottleWord,
                           bottles, bottleWord,
                           nextCount, nextBottleWord);
    }

    /**
     * Generates the song lyrics starting from a given number of bottles and for a specified number of verses.
     * @param startBottles The starting number of bottles (must be >= 0)
     * @param takeDown The number of verses to generate (must be >= 0 and <= startBottles + 1)
     * @return The song lyrics as a formatted string
     * @throws IllegalArgumentException if parameters are invalid
     */
    public String recite(int startBottles, int takeDown) {
        if (startBottles < 0) {
            throw new IllegalArgumentException("Starting bottles cannot be negative");
        }
        if (takeDown < 0) {
            throw new IllegalArgumentException("Number of verses cannot be negative");
        }
        if (takeDown > startBottles + 1) {
            throw new IllegalArgumentException("Cannot take down more bottles than available");
        }

        StringBuilder song = new StringBuilder();
        for (int i = startBottles; i > startBottles - takeDown; i--) {
            song.append(getVerse(i));
            if (i > startBottles - takeDown + 1) {
                song.append("\n");
            }
        }
        return song.toString();
    }
}
